bits.go

```diff
@@ -266,3 +267,69 @@ func fromDeltas(data []byte, buf []uint32) []uint32 {
 	}
 	return buf
 }
+
+type runeOffsetCorrection struct {
+	runeOffset, byteOffset uint32
+}
+
+// runeOffsetMap converts from rune offsets (with granularity runeOffsetFrequency)
+// to byte offsets, by tracking only the points where a span of runes is non-ASCII,
+// and otherwise interpolating expected byte offsets as one byte per rune.
+//
+// Instead of storing [100, 205, 305], it stores [{x: 200, y: 205}].
+//
+// This is very rarely a slight pessimization on repos where there are frequent
+// non-ASCII characters.
+type runeOffsetMap []runeOffsetCorrection
+
+// makeRuneOffsetMap converts the mostly-predictable runeOffset input
+// into a shorter form tracking the unexpected values.
+//
+// The input is a sequence of y values that we expect to increase by 100 each,
+// so we just store (x, y) points where the expectation is violated.
+func makeRuneOffsetMap(off []uint32) runeOffsetMap {
+	expected := uint32(0)
+	tmp := []runeOffsetCorrection{}
+	for runeOffset, byteOffset := range off {
+		if byteOffset != expected {
+			tmp = append(tmp, runeOffsetCorrection{uint32(runeOffset) * runeOffsetFrequency, byteOffset})
+			expected = byteOffset
+		}
+		expected += runeOffsetFrequency
+	}
+	// copy the slice to ensure it doesn't waste unused trailing capacity
```

Comment:
I think we can use the three index slicing here for the same effect without needing to copy. https://tip.golang.org/doc/go1.2#three_index